### DISCUSSION TITLE
Why does this action not cause an endless loop?

### DISCUSSION THREAD

**Discussion by preritdas** [stefanzweifel/git-auto-commit-action#238] [Help] [answered]
The workflow I've created is trigged on push. Because the git-auto-commit action (the final step of my workflow job) is pushing new files to my repository, I would expect it to cause an endless recursive loop of triggering the same workflow, as pushing causes a workflow that pushes, etc. 

I'm hoping someone can please explain why this doesn't happen in practice. I'm a bit of a newbie to GitHub Actions and workflows so I appreciate it in advance!

**Comment by stefanzweifel** [comment, marked as answer]
Good question which isn't obvious if you're just getting started with Actions.

To push the generated commit to GitHub, this Action uses the so called "[GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)". The token is used to authenticate with GitHub's APIs and allows this Action to push a commit to your repository.

When this Action pushes the commit to GitHub the `push`-event is triggered. However, if the event has been triggered with the GITHUB_TOKEN no further workflow run is triggered. Thus preventing an endless loop.

GitHub basically built in a failsafe, so you don't accidentally trigger a loop and burn through your Action minutes.

We've also mentioned this in the README [here](https://github.com/stefanzweifel/git-auto-commit-action#commits-made-by-this-action-do-not-trigger-new-workflow-runs). (You can disable this protection by using your own access token)

**Reply by preritdas** [reply]
Thanks a ton for the thorough and thoughtful response! I should have read the full README before asking this question. Appreciate you!

**Reply by stefanzweifel** [reply]
The README is way too long and contains so much information. It's easy to miss such details. :)
